packages/aws-cdk-lib/aws-appsync/lib/graphqlapi.ts

```diff
@@ -474,8 +570,19 @@ export class GraphqlApi extends GraphqlApiBase {
 
   /**
    * the schema attached to this api
+   * @deprecated Use apiSource.schema instead
+   */
+  public get schema(): ISchema {
+    if (this.apiSource.schema) {
+      return this.apiSource.schema;
+    }
+    throw new Error('Schema does not exist for AppSync merged APIs.');
+  }
+
+  /**
+   * API source (schema file or source APIs) for this GraphQL Api
    */
-  public readonly schema: ISchema;
+  public readonly apiSource: ApiSource;
```

Comment:
`schema` will still be there, it just got moved.

(But it will throw if the API was not created from a schema)

Reply:
Should `schema` be deprecated? I deprecated it in this PR. That means, it will be removed in CDK v3.
The schema will then be inaccessible, which could be considered a breaking change (information is no longer available). However, I am not sure how this property is used by other developers.